### PR TITLE
Tool changed event

### PR DESCRIPTION
**Pull request by Maxwell-Yang-PDFTron**
Implement Tool-changed event;
Updated some missing constants (or marked as TODO)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -611,6 +611,18 @@ array of `Config.ThumbnailFilterMode` tags, optional
 
 Defines filter modes that should be hidden in the thumbnails browser
 
+##### onToolChanged
+function, optional
+
+Defines what happens if there is a change to the current tool
```
Comment (JamesTron) [inline README.md:617]: I wouldn't use the word "Define", I'd say "notifies" or "is raised when" or whatever standard JS language is for when an event occurs.

**Review by JamesTron — COMMENTED** [review]
Minor change requested in documentation. Can merge without re-requesting review.

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -1665,6 +1785,35 @@ public void onAnnotationAction() {
...
+
+            String oldToolString = "";
+            if (oldTool != null) {
+                String toolString = convToolModeToString((ToolManager.ToolMode) oldTool.getToolMode());
+                if (toolString != null) {
+                    oldToolString = toolString;
+                }
+            }
+
+            WritableMap params = Arguments.createMap();
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:1808]: if (!Utils.isNullOrEmpty(newToolString)) {...} (only send event with valid new tool

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 3]
```diff
@@ -1789,26 +1789,22 @@ public void onAnnotationAction() {
...
+                oldToolString = convToolModeToString((ToolManager.ToolMode) newTool.getToolMode());
             }
 
+            String unknownString = "unknown tool";
+
             WritableMap params = Arguments.createMap();
             params.putString(ON_TOOL_CHANGED, ON_TOOL_CHANGED);
-            params.putString(KEY_previous_tool, oldToolString);
-            params.putString(KEY_tool, newToolString);
+            params.putString(KEY_previous_tool, oldToolString == null ? oldToolString : unknownString);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:1806]: ???

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java (hunk 3 above)
Comment (Maxwell-Yang-PDFTron) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:1806]: oops

File: README.md [hunk 4]
```diff
@@ -611,6 +611,18 @@ array of `Config.ThumbnailFilterMode` tags, optional
...
+##### onToolChanged
+function, optional
+
+This function is called when the current tool changes to a new tool
+
+Parameters:
+
+Name | Type | Description
+--- | --- | ---
+previousTool | string | the previous tool (one of the `Config.Tools` constants, or an empty string)
```
Comment (sgong-pdftron) [inline README.md:623]: update docs

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

**Pull request merged**